### PR TITLE
add rfcs for datafusion

### LINKED ISSUE #1397: Add RFCS for datafusion
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I think it's very good to have RFCs for datafusion. 

**Describe the solution you'd like**
Before developing a big and meaningful feature, it's better to have an RFC ticket. Maybe we used Google Docs to do the similar thing before, but if have an RFC ticket, we can discuss it easier and more conveniently, finally, we can merge the RFC to our codebase. These deposits will be a valuable part of datafusion. New developers will be easier to get to know datafusion.

**Describe alternatives you've considered**
RFC is also useful for developers to organize their thoughts, and reviewer reviews of code with an RFC will be easier

**Additional context**
such as https://github.com/cockroachdb/cockroach/tree/master/docs/RFCS
Later, I'll write an RFC about the subquery feature.

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

Closes #1397

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: docs/rfcs/template.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Authors:
+
+RFC PR: #
+
+Datafusion Issue: #
+
+---
+
+### BackGround
```
Comment (xudong963) [inline docs/rfcs/template.md:34]:
```suggestion
### Background
```

File: docs/rfcs/template.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+### General design
+
+---
+
+### Detailed design
+
+---
+
+### Questions
```
Comment (xudong963) [inline docs/rfcs/template.md:58]:
```suggestion
### Others
```

File: docs/source/community/communication.md [hunk 3]
```diff
@@ -76,9 +76,8 @@ Our source code is hosted on
 [GitHub](https://github.com/apache/arrow-datafusion). For developers new to
 the project, we have curated a
 [good-first-issue](https://github.com/apache/arrow-datafusion/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
-list to help you get started.
+list to help you get started. You can find datafusion's major designs in docs/rfcs.
```
Comment (alamb) [inline docs/source/community/communication.md:79]: I think the current specifications are actually in https://github.com/apache/arrow-datafusion/tree/master/docs/source/specification

**Review by alamb — COMMENTED** [review]
Thanks @xudong963

File: docs/source/community/communication.md (hunk 3 above)
Comment (xudong963) [reply docs/source/community/communication.md]: Yeah, I move RFCs into /source/specification

**Review by alamb — APPROVED** [review]
Looks reasonable to me. Thoughts @houqp  / @Jimexist  / @Dandandan ?

**Review by Jimexist — APPROVED** [review]

**Pull request merged**